projects/storefrontlib/src/cms-components/asm/services/asm-enabler.service.ts

```diff
@@ -79,7 +83,7 @@ export class AsmEnablerService {
     const factory = this.componentFactoryResolver.resolveComponentFactory(
       AsmMainUiComponent
     );
-    this.outletService.add('cx-storefront', factory, OutletPosition.BEFORE);
+    this.triggerService.renderDialog(this.triggerConfig.trigger?.asm, factory);
     this.isUiAdded = true;
```

Comment:
We should avoid the `isUiAdded` here. 
The maintenance of whether the UI was triggered already should be done in the trigger service. The trigger service will depending on the target, dialog type (dialog, message, notification, URL)  and (potentially) configuration decide whether it needs to be re-triggered.